plugin/src/main/scala/chisel3/internal/plugin/ChiselComponent.scala

```diff
@@ -21,6 +21,14 @@ class ChiselComponent(val global: Global, arguments: ChiselPluginArguments)
   class ChiselComponentPhase(prev: Phase) extends StdPhase(prev) {
     override def name: String = phaseName
     def apply(unit: CompilationUnit): Unit = {
+      // Issue a deprecation warning for Scala 2.12
+      val scalaVersion = scala.util.Properties.versionNumberString.split('.')
+      if (scalaVersion(0).toInt == 2 && scalaVersion(1).toInt == 12) {
+        val msg = s"Chisel 3.6 is the last version that will support Scala 2.12. Please upgrade to Scala 2.13."
+
+        global.reporter.warning(unit.body.pos, msg)
+      }
```

Comment:
```suggestion
      // Issue a deprecation warning for Scala 2.12
      val scalaVersion = scala.util.Properties.versionNumberString.split('.')
      if (scalaVersion(0).toInt == 2 && scalaVersion(1).toInt == 12) {
        val msg = s"Chisel 5 is the last version that will support Scala 2.12. Please upgrade to Scala 2.13."

        global.reporter.warning(unit.body.pos, msg)
      }
```
We're deprecating it in Chisel 5, not Chisel 3.6.

Also this warning prints for every single file which is sort of spammy, can you try to make it only warn once for the full project? I think this can by done by moving the logic here: https://github.com/chipsalliance/chisel/blob/3148a0e9f1dc6ccd85a80845ec513d03982440ac/plugin/src/main/scala/chisel3/internal/plugin/ChiselPlugin.scala#L31